http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/LeoAppServiceInterp.scala

```diff
@@ -806,6 +796,35 @@ final class LeoAppServiceInterp[F[_]: Parallel](config: AppServiceConfig,
       case (_, Some(gcpContext))   => F.pure[CloudContext](CloudContext.Gcp(gcpContext))
       case (None, None) => F.raiseError[CloudContext](CloudContextNotFoundException(workspaceId, ctx.traceId))
     }
+    // check if app can be deleted (Leo manages apps, so checking the leo status)
+    _ <- F.raiseUnless(app.status.isDeletable)(
+      AppCannotBeDeletedException(cloudContext, app.appName, app.status, ctx.traceId)
+    )
+
+    // check if databases, namespaces and managed identities associated with the app can be deleted
+    // (but only for Azure apps)
+    _ <-
+      if (cloudContext.cloudProvider == CloudProvider.Azure) {
+        for {
+          _ <- checkIfSubresourcesDeletable(app.id, WsmResourceType.AzureDatabase, userInfo, workspaceId)
+          _ <- checkIfSubresourcesDeletable(app.id, WsmResourceType.AzureManagedIdentity, userInfo, workspaceId)
+          _ <- checkIfSubresourcesDeletable(app.id, WsmResourceType.AzureKubernetesNamespace, userInfo, workspaceId)
+        } yield ()
+      } else F.unit
+
+    // Get the disk and check if its deletable (if disk is being deleted)
+    diskIdOpt = if (deleteDisk) app.appResources.disk.map(_.id) else None
+    _ = (deleteDisk, diskIdOpt, cloudContext.cloudProvider) match {
+      // only check WSM state for Azure apps (Azure apps don't have disks currently, but they are coming...)
+      case (true, Some(diskId), CloudProvider.Azure) =>
+        for {
+          _ <- checkIfSubresourcesDeletable(app.id, WsmResourceType.AzureDisk, userInfo, workspaceId)
```

Comment:
see comment above re: `checkIfSubresourcesDeletable(...)` before making the call.
thinking through this some more ... is this because we don't have a way of updating our database state that the resource was not deleted and are trying to provide feedback to the user sync vs sending the message and having them check the state at a later point-in-time?

Reply:
hmm maybe I'm not fully understanding this, but the idea behind checking if something is deletable in front leo is to fail faster so that the user can try again, rather than waiting until the message is processed by back leo, at which point it could be partially deleted and be in an unrecoverable state.

Reply:
yeah, I am seeing that now (getting more aligned with how things are currently designed). thank you for explaining this to me